src/components/orator/js/Orator.js

```diff
@@ -0,0 +1,963 @@
+     *******************************************************************************/
+
+    fluid.defaults("fluid.orator.domReader", {
+        gradeNames: ["fluid.viewComponent"],
+        selectors: {
+            highlight: ".flc-orator-highlight"
+        },
+        markup: {
+            highlight: "<mark class=\"flc-orator-highlight fl-orator-highlight\"></mark>"
+        },
+        events: {
+            onQueueSpeech: null,
+            onReadFromDOM: null,
+            utteranceOnEnd: null,
+            utteranceOnBoundary: null,
+            utteranceOnError: null,
+            utteranceOnMark: null,
+            utteranceOnPause: null,
+            utteranceOnResume: null,
+            utteranceOnStart: null
+        },
+        utteranceEventMap: {
+            onboundary: "utteranceOnBoundary",
+            onend: "utteranceOnEnd",
+            onerror: "utteranceOnError",
+            onmark:"utteranceOnMark",
+            onpause: "utteranceOnPause",
+            onresume: "utteranceOnResume",
+            onstart: "utteranceOnStart"
+        },
+        model: {
+            tts: {
+                paused: false,
+                speaking: false,
+                enabled: true
+            },
+            parseQueuelength: 0,
+            parseIndex: null,
+            ttsBoundary: null
+        },
+        modelRelay: [{
+            target: "parseIndex",
+            backward: "never",
+            namespace: "getClosestIndex",
+            singleTransform: {
+                type: "fluid.transforms.free",
+                func: "fluid.orator.domReader.getClosestIndex",
+                args: ["{that}", "{that}.model.ttsBoundary"]
+            }
+        }],
+        members: {
+            parseQueue: [],
+            range: {
+                expander: {
+                    this: "document",
+                    method: "createRange"
+                }
+            }
+        },
+        components: {
+            parser: {
+                type: "fluid.orator.domReader.parser"
+            }
+        },
+        invokers: {
+            readFromDOM: {
+                funcName: "fluid.orator.domReader.readFromDOM",
+                args: ["{that}", "{that}.container"]
+            },
+            removeHighlight: {
+                funcName: "fluid.orator.domReader.unWrap",
+                args: ["{that}.dom.highlight"]
+            },
+            setParseQueue: {
+                funcName: "fluid.orator.domReader.setParseQueue",
+                args: ["{that}", "{arguments}.0"]
+            },
+            resetParseQueue: {
+                funcName: "fluid.orator.domReader.setParseQueue",
+                args: ["{that}"]
+            },
+            highlight: {
+                funcName: "fluid.orator.domReader.highlight",
+                args: ["{that}", "{arguments}.0"]
+            },
+            play: {
+                funcName: "fluid.orator.domReader.speak",
```

Comment:
Awkward mismatch here between the name of the invoker and its implementing function.